cmd/create/accountroles/cmd.go

```diff
@@ -358,120 +360,144 @@ func run(cmd *cobra.Command, argv []string) {
 	}
 }
 
-func buildCommands(prefix string, permissionsBoundary string, accountID string, defaultPolicyVersion string,
-	path string, managedPolicies bool, policies map[string]*cmv1.AWSSTSPolicy) (string, error) {
-	commands := []string{}
+type creator interface {
+	createRoles(*rosa.Runtime, rolesCreationInput) error
+	getRoleTags(string, rolesCreationInput) map[string]string
+	buildCommands(rolesCreationInput) (string, error)
+}
+
+func initCreator(managedPolicies bool) creator {
+	if managedPolicies {
+		return managedPoliciesCreator{}
+	}
+
+	return unmanagedPoliciesCreator{}
+}
+
+func buildRolesCreationInput(prefix, permissionsBoundary, accountID, env string,
+	policies map[string]*cmv1.AWSSTSPolicy, defaultPolicyVersion string,
+	path string) rolesCreationInput {
+	return rolesCreationInput{
+		prefix:               prefix,
+		permissionsBoundary:  permissionsBoundary,
+		accountID:            accountID,
+		env:                  env,
+		policies:             policies,
+		defaultPolicyVersion: defaultPolicyVersion,
+		path:                 path,
+	}
+}
+
+type rolesCreationInput struct {
```

Comment:
I know this is only used here and that it is internal, but what do you think about naming it AccountRolesCreationInput? Just so we don't mix it up with other structs that will show up from refactors

Reply:
Changed.